src/design-system/InputToggle.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import styled from "styled-components";
+
+import Colors from "./Colors";
+
+interface Props {
+  toggled?: boolean;
+  onChange: (e: React.ChangeEvent) => void;
+}
+
+const ToggleContainer = styled.div`
+  display: flex;
+  flex-direction: row nowrap;
+`;
+
+const Label = styled.label<{ toggled?: boolean }>`
+  align-self: center;
+  color: ${(props) => (props.toggled ? Colors.teal : Colors.darkTeal)};
+  font-size: 14px;
+  font-family: "Poppins", sans-serif;
+  min-width: 50px;
+  padding: 0 1em;
+`;
+
+const ToggleInput = styled.input`
+  height: 0;
+  visibility: hidden;
+  width: 0;
+`;
+
+const ToggleButtonContainer = styled.label<{ toggled?: boolean }>`
+  align-items: center;
+  background: ${(props) => (props.toggled ? Colors.teal : Colors.darkTeal)};
+  border-radius: 20px;
+  cursor: pointer;
+  display: flex;
+  height: 26px;
+  justify-content: space-between;
+  position: relative;
+  transition: background-color 0.2s;
+  width: 48px;
+`;
+
+const ToggleButton = styled.span<{ toggled?: boolean }>`
+  background: ${Colors.slate};
+  border-radius: 50%;
+  content: "";
+  cursor: pointer;
+  height: 15px;
+  left: ${(props) => (props.toggled ? "calc(100% - 5px)" : "5px")};
+  position: absolute;
+  top: 5px;
+  transition: 0.2s;
+  ${(props) => (props.toggled ? `transform: translateX(-100%);` : null)}
+  width: 15px
+`;
+
+const InputToggle: React.FC<Props> = (props) => {
+  const { toggled, onChange } = props;
+  const label = toggled ? "On" : "Off";
+
+  return (
+    <ToggleContainer>
+      <Label toggled={toggled}>{label}</Label>
+      <ToggleInput
+        id="ToggleInput"
```

Comment:
I'm not 100% sure about this (not super experienced on the frontend), but if this becomes an id on the rendered HTML element and there are many of these components on the page, then they'd all share the id and we may have issues with any code that tries to identify a given input in the document by id. Do we need this id field or can we just omit it?

Reply:
good callout ... it looks like we do need it, @daschi i would recommend just augmenting it with `lodash.uniqueId` since we already have that package as a dependency

Reply:
Oh yes, great point--I'll add that!